common/src/main/kotlin/com/hartwig/actin/icd/IcdModel.kt

```diff
@@ -78,6 +78,12 @@ class IcdModel(
         )
     }
 
+    fun <T: Comorbidity> findInstancesMatchingAnyExtensionCode(instances: List<T>, targetExtensionCodes: Set<String>): List<Comorbidity> {
+        return instances.filter { entity ->
+            entity.icdCodes.flatMap { codeWithAllParents(it.extensionCode) }.toSet().any(targetExtensionCodes::contains)
```

Comment:
I think you could do this without building intermediate collections:
```suggestion
            entity.icdCodes.any { codeWithAllParents(it.extensionCode).any(targetExtensionCodes::contains) }
```